docs/guides/integration-dsp-no-sdk.md

```diff
@@ -0,0 +1,160 @@
+
+For more information about client-side UID2 integration, refer to one of these integration guides:
+
+- [Client-Side Integration Guide for JavaScript](publisher-client-side.md)
+- [UID2 Client-Side Integration Guide for Prebid.js](integration-prebid-client-side.md)
+- [UID2 Client-Side Integration Guide for Mobile](integration-mobile-client-side.md)
+
+
+### Make Sure Token Lifetime and Expiration Are Valid
+
+A token must be valid and current so that it can be used in the bidstream. You must do two things:
+
+- Make sure that the token hasn't expired.
+- Check that the token lifetime is a valid value.
+
+For an example of code that makes sure the token hasn't expired, see [UID2Encryption.cs, line 196](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L196).
+
+To make sure that the token lifetime has a valid value, check these two conditions:
+
+- The lifetime of the token must not exceed the `max_bidstream_lifetime_seconds` value from the `/v2/key/bidstream` response.
+- The time until token generation value must not exceed the `allow_clock_skew_seconds` value.
+
+ For an example of how this is done, review the code for the `DoesTokenHaveValidLifetimeImpl` function: see [UID2Encryption.cs, line 237](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L237).
+
+The following sections show how the lifetime for a token is calculated, and the time until token generation for versions later than v2. The calculation depends on the token version:
+
+- [Calculating Token Lifetime: Token v2](#calculating-token-lifetime-token-v2)
+- [Calculating Token Lifetime: All Later Versions](#calculating-token-lifetime-all-later-versions)
+
+#### Calculating Token Lifetime: Token v2
+
+For token v2, the calculation to make sure that the token lifetime is valid for bidstream use is as follows:
+
+```
+lifetime = token expiry - current time
+```
+
+For v2, we use the token expiry minus the current time to calculate the lifetime. This is because v2 doesn't have a **Token Generated** field, which is present in later versions. All token versions have an **Identity Established** field, but this indicates the time that the original token was generated, before any token refreshes, so it can't be used to calculate whether the token is still valid.
+
+#### Calculating Token Lifetime: All Later Versions
+
+For all token versions later than v2, the calculation to make sure that the token lifetime is valid for bidstream use is as follows:
+
+```
+lifetime = token expiry - token generated
+
+time until token generation = token generated - current time
+```
+
+Versions later than v2 have a **Token Generated** field, which is updated if the token is refreshed, so we use this to calculate the token lifetime.
+
+### Conditional: Verify the Domain or App Name
```

Comment:
I suggest removing the Conditional wording and follow the similar pattern as above and change to 

_Verify the Domain or App Name for a Token Generated on the Client Side_

Reply:
Done, see my note in Slack for updated visuals.